mesh/submesh/transfermap.cpp

```diff
@@ -92,10 +92,13 @@ TransferMap::TransferMap(const GridFunction &src,
 
          if (!root_fes_reset)
          {
+            const FiniteElementCollection *src_fec = src.FESpace()->FEColl();
+            const FiniteElementCollection *dst_fec = dst.FESpace()->FEColl();
+            auto *root_fec = src_sm_dim == parent_dim ? src_fec : dst_fec;
```

Comment:
For posterity: this relies on the assumption that the intersection of two d-1 submeshes must be an empty set. This isn't strictly true topologically, as it could be a d-2 object (two planes meeting at a surface, in a diamond dependency pattern), but the common root mesh between two submeshes would still be a d-mesh, as `dim(parent) >= dim(child)`.

Reply:
Hi, 

Do you mind elaborating on this a bit? I'm having difficulty following.

 - When you say d, is that the topological dimension of the parent mesh? 
 - If the intersection must be an empty set then the submeshes can only have a common boundary? Why can't there be an intersection?
 - If both submeshes have dim lower than the parent mesh this will default to the destination mesh; does this matter (is it preferred to use the source fec?)
 
 Thanks!

Reply:
* When I say a d-mesh I mean a d-dimensional mesh, so a mesh made up of hexes is a 3-mesh. 
* Intersection here isn't quite correct, as you say the submeshes might both be made of overlapping sets of attributes, {0,1} and {1,2} for instance. If two d-1 meshes had a d-2 intersection, then solution transferring doesn't make sense, hence excluding it (that's the bit in parentheses). 
* However, if both meshes have a lower-dimension, taking 3D as example, that means transferring a solution between two surfaces (2-meshes) then the only way they could transfer between, is if the two surfaces overlapped: the 2-measure of the intersection is not zero. The only way that the solution could be transferred from the source to the destination would then be if the destination is a strict superset of the source (so transfering from {0} to {0,1} boundary attributes). I don't think the choice of fec makes too much of a difference here though, as it's assuming that one fec can map to another without loss, so probably could've picked either. The bigger aim was for the case of picking the higher dimensional of source and destination as the fec for d-1 -> d transfer.